### PR TITLE
FIX: concatenate_epochs now checks for event_id correspondence

### LINKED ISSUE #3992: concatenate_epochs should check event_id values correspondence
I've recently had a weird bug - concatenating two epochs led to assertion error in epoch constructor. After some debugging I found out that the cause of the problem was that the epochs did not have the same event_id values (the same key corresponded to different values in these files). [This line in `concatenate_epochs`](https://github.com/mne-tools/mne-python/blob/master/mne/epochs.py#L2708) assumes that same keys correspond to same values and uses `update` method dict method to join dicts. It would be better if concatenate_epochs checked if this is true and threw error otherwise.

### PR DESCRIPTION
**Pull request by mmagnuski**
Fixes #3992

I'll add a note to whats new once CI's are green.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -2700,6 +2700,14 @@ def _concatenate_epochs(epochs_list, with_data=True):
         if epochs.baseline != baseline:
             raise ValueError('Baseline must be same for all epochs')
 
+        # compare event_id
+        common_keys = list(set(event_id.keys()).intersection(
```
Comment (agramfort) [inline mne/epochs.py]: why do you need the .keys()?

Comment (agramfort) [inline mne/epochs.py]: it will not work with event_id been an integer

Comment (agramfort) [inline mne/epochs.py]: also I think that the 

event_id  == epochs.event_id

should do the job

Comment (mmagnuski) [inline mne/epochs.py]: oh, true with event_id not being dict, I forgot about this.
Regarding your last comment - I don't think concatenating epochs with different event_id keys should be forbidden ie. `concatenate_epochs([epochs['condition1'], epochs['condition2'])` should work. 
Unless different event_id keys map to the same value... I may also add a check for this if it is needed.

Comment (mmagnuski) [inline mne/epochs.py]: I'll remove `.keys()` too.

Comment (mmagnuski) [inline mne/epochs.py]: actually epochs.event_id should always be dict, right? With `epochs = mne.Epochs(..., event_id=1)` you get `epochs.event_id == {'1': 1}`.

**Comment by agramfort** [comment]
hum indeed.

Also I realize that doing epochs['condition1'] should then restrict the
event_id to the matching conditions...

File: mne/epochs.py [hunk 2]
```diff
@@ -2700,6 +2700,13 @@ def _concatenate_epochs(epochs_list, with_data=True):
         if epochs.baseline != baseline:
             raise ValueError('Baseline must be same for all epochs')
 
+        # compare event_id
+        common_keys = list(set(event_id).intersection(set(epochs.event_id)))
+        for key in common_keys:
+            if not event_id[key] == epochs.event_id[key]:
+                raise ValueError('event_id values must be the same for '
+                                 'identical keys for all concatenated epochs.')
```
Comment (agramfort) [inline mne/epochs.py]: say what are different to make the error explicit.

Comment (agramfort) [inline mne/epochs.py]: besides +1 for MRG

**Comment by agramfort** [comment]
thx @mmagnuski

**Comment by mmagnuski** [comment]
Thanks @agramfort ! 🚀

**Comment by mmagnuski** [comment]
@agramfort 
> Also I realize that doing epochs['condition1'] should then restrict the
event_id to the matching conditions...

I thought it does - if not it is probably an issue/PR material.

**Pull request merged**